app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,103 @@
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
+
+/**
+ * [RecyclerView.ViewHolder] for displaying an individual [TabCollection].
+ * Clients are expected to use [bindSession] to link a particular [TabCollection] to be displayed
+ * otherwise this will be an empty, 0 size View.
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param interactor [CollectionInteractor] callback for user interactions.
+ */
+class CollectionViewHolder(
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
+    private val interactor: CollectionInteractor,
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
+    private var collectionData = CollectionInfo()
+
+    @Composable
+    override fun Content() {
+        val collectionInfo by remember { mutableStateOf(collectionData) }
+
+        collectionInfo.collection?.let { validCollection ->
+            val horizontalPadding =
+                composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+            composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
+
+            val menuItems = getMenuItems(
+                collection = validCollection,
+                onCollectionOpenTabsTapped = interactor::onCollectionOpenTabsTapped,
+                onRenameCollectionTapped = interactor::onRenameCollectionTapped,
+                onCollectionAddTabTapped = interactor::onCollectionAddTabTapped,
+                onDeleteCollectionTapped = interactor::onDeleteCollectionTapped,
+            )
+
+            Column {
+                Spacer(Modifier.height(12.dp))
+
+                Collection(
+                    collection = validCollection,
+                    expanded = collectionInfo.isExpanded,
+                    menuItems = menuItems,
+                    onToggleCollectionExpanded = interactor::onToggleCollectionExpanded,
+                    onCollectionShareTabsClicked = interactor::onCollectionShareTabsClicked,
+                    onCollectionMenuOpened = interactor::onCollectionMenuOpened,
+                )
+            }
+        }
+    }
+
+    /**
+     * Dynamically replace the current [TabCollection] shown in this `ViewHolder`.
+     *
+     * @param collection [TabCollection] to be shown
+     * @param expanded Whether to show the collection as expanded or collapsed.
+     */
+    fun bindSession(collection: TabCollection, expanded: Boolean) {
+        collectionData = CollectionInfo(
+            collection = collection,
+            isExpanded = expanded
+        )
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+        const val maxTitleLength = 20
```

Comment:
Can we move this out to where it's being called in `TabCollectionStorage` as a constant? Considering that the days for keeping `CollectionViewHolder` is probably limited we are probably prefer serve to refactor it now. I only bring this refactoring up because the review UI makes this file as appear as entirely new, even though I know we refactored it correctly.